Products/CMFPlomino/PlominoForm.py

```diff
@@ -1242,11 +1242,14 @@ def readInputs(self, doc, REQUEST, process_attachments=False, applyhidewhen=True
                                 doc,
                                 process_attachments,
                                 validation_mode=validation_mode)
-                        if f.getFieldType() == 'SELECTION':
-                            if f.getSettings().widget in [
-                                    'MULTISELECT', 'CHECKBOX', 'PICKLIST']:
-                                v = asList(v)
-                        doc.setItem(fieldName, v)
+                        if v != None:
+                            if f.getFieldType() == 'SELECTION':
+                                if f.getSettings().widget in [
+                                        'MULTISELECT', 'CHECKBOX', 'PICKLIST']:
+                                    v = asList(v)
+                            doc.setItem(fieldName, v)
+                        else:
+                            doc.removeItem(fieldName)
```

Comment:
Need a comment to specify this can happen in the case of datetimes which need to process the input first before its possible to work out they are empty. 
Also are you sure the old code removed the item if you went and edited it and reset it all back to 00?